geode-core/src/main/java/org/apache/geode/internal/cache/PoolFactoryImpl.java

```diff
@@ -339,15 +340,36 @@ public Pool create(String name) throws CacheException {
     InternalDistributedSystem distributedSystem = InternalDistributedSystem.getAnyInstance();
     InternalCache cache = GemFireCacheImpl.getInstance();
     ThreadsMonitoring threadMonitoring = null;
+
     if (cache != null) {
       threadMonitoring = cache.getDistributionManager().getThreadMonitoring();
       TypeRegistry registry = cache.getPdxRegistry();
       if (registry != null && !attributes.isGateway()) {
         registry.creatingPool();
       }
     }
-    return PoolImpl.create(this.pm, name, this.attributes, this.locatorAddresses, distributedSystem,
-        cache, threadMonitoring);
+
+    Pool returnPool = null;
+    int retries = 0;
+    final int MAX_RETRIES = 20;
+    while (returnPool == null && retries < MAX_RETRIES) {
+      try {
+        returnPool =
+            PoolImpl
+                .create(this.pm, name, this.attributes, this.locatorAddresses, distributedSystem,
+                    cache, threadMonitoring);
+        logger.info(" **** Did we retry?");
+      } catch (NoSubscriptionServersAvailableException nsa) {
+        logger.warn("About to retry");
+        retries++;
+        try {
+          Thread.sleep(100);
+        } catch (InterruptedException e) {
```

Comment:
The contract for interrupt in Java is to never catch InterruptedException without resetting the interrupt flag. Also if the code is written such that resetting the interrupt flag breaks it, then the approach of the code is not correct.
```
try {
  Thread.sleep(100);
} catch (InterruptedException e) {
  Thread.currentThread().interrupt();
}
```
Which admittedly doesn't look good so the code probably needs some further changes. Grep for 'boolean interrupted' and you'll find lots of examples of code that's using interrupts correctly (or at least more correctly).